### ISSUE TITLE
Optional function parameters

### ISSUE THREAD

**Issue by congysu** [OData/odata.net#403]
Such as:

```
// Supported now.
func(a='hello', b='world')
func(a='hello', b=null)
func(a='hello', b=)

// Not supported for now.
func(a='hello')
```

without creating overloads, if `b` is nullable.

**Comment by jatin085** [comment]
Adding a note: the third syntax does not seem to work `func(a='hello', b=)` which is okay as it's not part of OData specification.

**Comment by mikepizzo** [comment]
Fixed by PR #828.
